doc/README.md

```diff
@@ -0,0 +1,67 @@
+## Adding GRPC Support to Tailcall Overview
+
+This proposal outlines an approach for adding GRPC support to Tailcall to allow generating GRPC server stubs from GraphQL schemas.
+
+
+# Design Goals
+
+The primary design goals are:
+
+Allow configuring GRPC servers alongside HTTP servers in Tailcall config
+Generate GRPC server stubs from GraphQL schemas
+Enable handling GRPC requests with Tailcall resolvers
+Integrate cleanly with existing Tailcall architecture
+Proposed Solution
+The proposed solution is to:
+
+Add a grpc section to the Tailcall config file, similar to http:
+
+```bash
+grpc:
+  port: 50051
+```
+
+
+Add a @grpc directive to mark GraphQL types as GRPC messages:
+
+```bash
+@grpc
+
+type User {
+  id: ID! 
+  name: String!
+}
+```
+
+
+Generate GRPC proto files from GraphQL schemas with @grpc types.
+Use a GRPC code generation tool like grpc-rust to generate GRPC server stubs from the proto files.
+Start a GRPC server alongside the HTTP server to handle requests.
+Convert between GRPC requests/responses and GraphQL types to reuse existing Tailcall resolvers.
+
+Implementation Details
+
+The @grpc directive would mark a GraphQL object type as a GRPC message.
+The proto file generator would output .proto files from the GraphQL schema, containing any @grpc types as messages.
```

Comment:
Do we need to create proto files? Can't we keep them in memory? Please expand on where the protobufs be stored.

Reply:
The .proto files are meant to be a static definition of your data structure and services right. Though we could do something as follows:
Define Protobufs -> Generate .proto Files -> Compile .proto Files -> Use Generated Code -> Validate Responses